packages/node_modules/@webex/internal-plugin-ediscovery/src/index.js

```diff
@@ -183,58 +228,106 @@ registerInternalPlugin('ediscovery', EDiscovery, {
       {
         name: 'decryptReportContent',
         direction: 'inbound',
-        fn(ctx, object) {
+        fn(ctx, object, reportId = '') {
           if (!object) {
             return Promise.resolve();
           }
 
           const promises = [];
-          if (object.verb === 'post' && object.objectDisplayName && object.encryptionKeyUrl) {
-            promises.push(ctx.spark.internal.encryption.decryptText(object.encryptionKeyUrl, object.objectDisplayName, {onBehalfOf: object.actorId})
-              .then((decryptedMessage) => {
-                object.objectDisplayName = decryptedMessage;
-              })
-              .catch((reason) => {
-                ctx.spark.logger.info(`Decrypt message error for activity ${object.activityId}: ${reason}`);
-                return object; // TODO - determine correct behaviour when an individual operation fails
-              }));
-          }
-          else if (object.verb === 'share' && object.files && object.files.length && object.encryptionKeyUrl) {
-            for (let i = 0; i < object.files.length; i += 1) {
-              const file = object.files[i];
-              if (file && file.displayName && file.scr) {
-                promises.push(ctx.spark.internal.encryption.decryptText(object.encryptionKeyUrl, file.displayName, {onBehalfOf: object.actorId})
-                  .then((decryptedFilename) => {
-                    file.name = decryptedFilename;
+          return ctx.spark.internal.ediscovery._getContentSummaryBySpaceId(reportId, object.targetId)
+            .then((spaceSummary) => {
+              const spaceParticipantId = spaceSummary.participants[0] || object.actorId;
```

Comment:
Use participant if possible, as actorId may have left the space since activity was created